### PR TITLE
diagnostics for debugging Python and SymPy installs

### PR DESCRIPTION
**Pull request by cbm755**
@mtmiller want to collaborate on exactly what this might look like?

Here's my draft.  Its more verbose but not a lot of new info yet.

Having done this, I'm thinking it could just be `assert_have_python_and_sympy` with a `verbose` flag.

### CODE REVIEW

**Comment by mtmiller** [comment]
Great, looks good to me so far.

Agree with combining with `assert_have_python_and_sympy`. This would serve a few useful purposes: reducing duplication of code, no risk of having the tests in this function not matching the actual requirements used when setting up the interface, and it would be a sort of self-documenting annotation showing what the point of each of the tests is.

File: inst/private/assert_have_python_and_sympy.m [hunk 1]
```diff
@@ -16,35 +16,117 @@
...
-function assert_have_python_and_sympy(pyexec)
-%private function
+%% -*- texinfo -*-
+%% @defun assert_have_python_and_sympy (pyexec, verbose)
+%% Check that Python and SymPy are installed and working.
+%%
+%% @seealso{sympref}
+%% @end defun
+
+function assert_have_python_and_sympy (pyexec, verbose)
```
Comment (mtmiller) [inline inst/private/assert_have_python_and_sympy.m]: `verbose = false`? Does Matlab allow default arg values?

Or `if (nargin < 2) verbose = false; end`

**Comment by cbm755** [comment]
AFAIK, Matlab does not (did not?) allow default arg values.

1.  We could list `PYTHONPATH` env, although surely anyone messing with that is not the target audience...

2.  In the case when sympy is found but is too old, maybe we could give some info about the python we found (https://stackoverflow.com/questions/247770/retrieving-python-module-path/12154601#12154601)

3.  OS information?  Ubuntu version?  "uname -a" output if on unix?

Or maybe its good enough for now?

**Comment by mtmiller** [comment]
Yeah I think the tests performed are good enough for now.

The only change I would make now is maybe format the "suggestions" part of the output better so they stand out. The most important feature of this is giving the user clear actions to try next.

Instead of this block of text

    Unfortunately status was non-zero: probably Python cannot import sympy.
    Have you installed SymPy?  Please try to install it and try again.
    If you have installed SymPy, perhaps for a different Python interpreter?
    Please try "setenv" as described above to change your python interpreter.

maybe space it out and itemize it?

    Unfortunately status was non-zero: probably Python cannot import sympy.

    * Do you have SymPy installed?  If not, please try to install it and try again.

    * If you do have SymPy installed, maybe it's installed for a different Python interpreter than the one we found?
      Please try "setenv" as described above to change your python interpreter.

**Comment by mtmiller** [comment]
Other possible future enhancements:
* bold / color emphasis
* machine-readable output with `result = sympref ('diagnose')`
* a place to try multiple pythons, as in #771

File: inst/private/assert_have_python_and_sympy.m [hunk 2]
```diff
@@ -16,35 +16,126 @@
...
+    disp ('Symbolic package diagnostics')
+    disp ('============================')
+    disp ('')
+    disp ('Python and SymPy are needed for most features of the Symbolic package.')
+    disp ('')
+    fprintf ('The Python interpreter is currently: "%s".\n', pyexec)
+    disp ('')
+    disp ('Computers may have more than one Python interpreter installed.  Depending')
+    disp ('on how your system is configured, you may need to select a different one.')
+    disp ('See "help sympref" for details, but for example, to use Python 3, try')
```
Comment (cbm755) [inline inst/private/assert_have_python_and_sympy.m]: I will try to rephrase this: sounds like something is wrong and action is required.

**Comment by cbm755** [comment]
I like your suggestions.   Will do those and merge (leaving future enhancement for the future!)

**Pull request merged**